src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -299,6 +299,24 @@ private function parseDefinition($id, $service, $file)
             $definition->setDecoratedService($service['decorates'], $renameId, $priority);
         }
 
+        if (isset($service['autowire'])) {
+            $definition->setAutowired($service['autowire']);
+        }
+
+        if (isset($service['autowiring_types'])) {
+            if (!is_array($service['autowiring_types'])) {
+                throw new InvalidArgumentException(sprintf('Parameter "autowiring_types" must be an array for service "%s" in %s. Check your YAML syntax.', $id, $file));
```

Comment:
As having only one type is quite common, I would allow a single string to be passed here.